builder/simplify.cc

```diff
@@ -24,7 +24,16 @@ namespace {
 
 // This is based on the simplifier in Halide: https://github.com/halide/Halide/blob/main/src/Simplify_Internal.h
 class simplifier : public node_mutator {
-  symbol_map<int> references;
+  class reference_info {
```

Comment:
Naming nits:
- `reference_info` -> `symbol_info` (this struct describes many references for a symbol, not just one reference)
- Remove trailing underscores from public members
- `counter_` -> `ref_count`?
- `present_in_loop_` -> `referenced_in_loop`?